providers/teams/pages/members/[id].tsx

```diff
@@ -0,0 +1,215 @@
+query getUsers($userMail: String!) {
+    users(where: {email: {_eq: $userMail }}) {
+        id
+    }
+}
+`
+
+
+const hasTeam = gql`
+query hasTeam($userId: Int!) {
+    team_members(where: { userId: { _eq: $userId } }) {
+        teamId
+    }
+}
+`
+
+const insertTeamMember = gql`
+mutation InsertTeamMembers($objects: [team_members_insert_input!]!) {
+    insert_team_members(objects: $objects) {
+      affected_rows
+    }
+  }
+`
+
+
+const deleteTeamMember = gql`
+mutation DeleteTeamMembers($uid: Int!, $tid: Int!) {
+    delete_team_members(where: {_and: {userId: {_eq: $uid}, teamId: {_eq: $tid}}}) {
+      returning {
+        user {
+          id
+          email
+        }
+        teamId
+      }
+    }
+  }
+`
+
+async function checkHasTeam(userId: string) {
+    const graphqlMutation = await apolloClient.query({
+        query: hasTeam,
+        variables: { userId }
+    })
+
+    if (graphqlMutation.data.team_members && graphqlMutation.data.team_members.length) {
+        return graphqlMutation.data.team_members[0].teamId
+    }
+
+    return null
+}
+
+
+const addMember = async (teamId: string, userMail: string) => {
+    const { loading, data, errors } = await apolloClient.query({
+        query: getUserByMail,
+        variables: { userMail }
+    })
+    if (loading) return <div>Loading...</div>
+    if (errors || !data) console.error(errors || "Couldn't retrieve data.")
+
+    if (data.users.length > 0) {
+        const userId = data.users[0].id
+        console.log(`userId ${userId}`);
+
+
+        const hasTeamId = await checkHasTeam(userId)
+        if (hasTeamId) {
+            console.log(`${userMail} is already in team ${hasTeamId}`);
+        } else {
+            // add user to team
+            const { data, errors } = await apolloClient.mutate({
+                mutation: insertTeamMember,
+                variables: { objects: { teamId: parseInt(teamId), userId } }
+            })
+            if (errors) {
+                alert(errors[0])
+                throw errors[0]
+            }
+            return 'success'
+        }
+    } else {
+        // user not found
+        return null
+    }
+}
+
+
+const removeMember = async (memberData) => {
+    // remove user from team
+    console.log('removing user')
+    console.log(memberData);
+
+    const { data, errors } = await apolloClient.mutate({
+        mutation: deleteTeamMember,
+        variables: { uid: memberData.user.id, tid: memberData.teamId }
+    })
+    console.log(data)
+    console.log(data.delete_team_members.returning);
+    if (errors) {
+        alert(errors[0])
+        throw errors[0]
+    }
+    return 'success'
+}
+
+
+
+export const TeamEditMembersPage = () => {
+
+
+    const router = useRouter();
+    const [teamId, setTeamId] = useState((typeof router.query.id === 'string') ? router.query.id : router.query.id[0]);
+    const [data, setData] = useState(null)
+    const [teamMembers, setTeamMembers] = useState([]);
```

Comment:
Would you like to try and imitate `useQuery` from `TeamProfilePage`? it would supposedly know when you changed the page and reload the data accordingly

Reply:
I tried,
it caused problems with useEffect.
I think it had something to do with next.js pre-rendering but I'm not sure..
The problem was that the app failed after I refreshed due to a "change in the number of hooks"
I'll keep it this way and maybe I'll try to use useQuery later down the road